From 5c3881bcdcfa63466100755dc3c8a5f34b647f67 Mon Sep 17 00:00:00 2001
From: lyctw <lyctw.ee@gmail.com>
Date: Mon, 4 Oct 2021 15:34:42 +0800
Subject: [PATCH 1/3] Solve mmc no partition table error

Signed-off-by: lyctw <lyctw.ee@gmail.com>
---
 drivers/mmc/ftsdc010_mci.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/mmc/ftsdc010_mci.c b/drivers/mmc/ftsdc010_mci.c
index bc0d5ffe..8d1a84e5 100644
--- a/drivers/mmc/ftsdc010_mci.c
+++ b/drivers/mmc/ftsdc010_mci.c
@@ -444,9 +444,9 @@ static int ftsdc010_mmc_probe(struct udevice *dev)
 		return ret;
 #endif
 
-	if (dev_read_bool(dev, "cap-mmc-highspeed") || \
-		  dev_read_bool(dev, "cap-sd-highspeed"))
-		chip->caps |= MMC_MODE_HS | MMC_MODE_HS_52MHz;
+	//if (dev_read_bool(dev, "cap-mmc-highspeed") || \
+	//	  dev_read_bool(dev, "cap-sd-highspeed"))
+	//	chip->caps |= MMC_MODE_HS | MMC_MODE_HS_52MHz;
 
 	ftsdc_setup_cfg(&plat->cfg, dev->name, chip->buswidth, chip->caps,
 			priv->minmax[1] , priv->minmax[0]);
-- 
2.17.0

