From 8e860521bcb3813fe18988afd8661d234bbccd68 Mon Sep 17 00:00:00 2001
From: Yu Chien Peter Lin <peterlin@andestech.com>
Date: Wed, 29 Dec 2021 13:34:48 +0800
Subject: [PATCH 3/3] Enable prints during boot

Signed-off-by: Yu Chien Peter Lin <peterlin@andestech.com>
---
 include/opensbi.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/opensbi.h b/include/opensbi.h
index d812cc8c..3121355e 100644
--- a/include/opensbi.h
+++ b/include/opensbi.h
@@ -20,7 +20,7 @@

 enum sbi_scratch_options {
	/** Disable prints during boot */
-	SBI_SCRATCH_NO_BOOT_PRINTS = (1 << 0),
+    SBI_SCRATCH_NO_BOOT_PRINTS = (0),
 };

 /** Representation dynamic info passed by previous booting stage */
--
2.25.1

