From ce997b346b90edc37b630a98b090d654f89d0ec1 Mon Sep 17 00:00:00 2001
From: Yu Chien Peter Lin <peterlin@andestech.com>
Date: Thu, 16 Feb 2023 11:12:00 +0800
Subject: [PATCH] force flush pipeline for outstanding memory access

Signed-off-by: Yu Chien Peter Lin <peterlin@andestech.com>
---
 lib/sbi/riscv_locks.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/sbi/riscv_locks.c b/lib/sbi/riscv_locks.c
index acab776..218996e 100644
--- a/lib/sbi/riscv_locks.c
+++ b/lib/sbi/riscv_locks.c
@@ -52,6 +52,7 @@ void spin_lock(spinlock_t *lock)
 	u32 l0, tmp1, tmp2;
 
 	__asm__ __volatile__(
+		"fence\n"
 		/* Atomically increment the next ticket. */
 		"	amoadd.w.aqrl	%0, %4, %3\n"
 
-- 
2.34.1

