From be4dbc351df5251eebc9f5c8940de42c13f92f3c Mon Sep 17 00:00:00 2001
From: Yu Chien Peter Lin <peterlin@andestech.com>
Date: Tue, 16 Aug 2022 18:11:14 +0800
Subject: [PATCH] sbi: riscv_locks.c: memory barrior for AMO instruction

Signed-off-by: Yu Chien Peter Lin <peterlin@andestech.com>
---
 lib/sbi/riscv_locks.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/sbi/riscv_locks.c b/lib/sbi/riscv_locks.c
index acab776..218996e 100644
--- a/lib/sbi/riscv_locks.c
+++ b/lib/sbi/riscv_locks.c
@@ -52,6 +52,7 @@ void spin_lock(spinlock_t *lock)
 	u32 l0, tmp1, tmp2;
 
 	__asm__ __volatile__(
+		"fence\n"
 		/* Atomically increment the next ticket. */
 		"	amoadd.w.aqrl	%0, %4, %3\n"
 
-- 
2.34.1

