From 85ee54a6eea99d2a82ea182d3281e67ce13d76b3 Mon Sep 17 00:00:00 2001
From: Yu Chien Peter Lin <peterlin@andestech.com>
Date: Tue, 2 Nov 2021 15:40:24 +0800
Subject: [PATCH] Fix gcc 11 compiling errors

Signed-off-by: Yu Chien Peter Lin <peterlin@andestech.com>
---
 Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 7f43b92..a8c7696 100644
--- a/Makefile
+++ b/Makefile
@@ -194,6 +194,7 @@ GENFLAGS	+=	$(platform-genflags-y)
 GENFLAGS	+=	$(firmware-genflags-y)
 
 CFLAGS		=	-g -Wall -Werror -ffreestanding -nostdlib -fno-stack-protector -fno-strict-aliasing -O2
+CFLAGS		+=	-Wno-stringop-overflow
 CFLAGS		+=	-fno-omit-frame-pointer -fno-optimize-sibling-calls
 CFLAGS		+=	-mno-save-restore -mstrict-align
 CFLAGS		+=	-mabi=$(PLATFORM_RISCV_ABI) -march=$(PLATFORM_RISCV_ISA)
@@ -219,7 +220,7 @@ ASFLAGS   +=  -fno-pie -no-pie
 
 ARFLAGS		=	rcs
 
-ELFFLAGS	+=	-Wl,--build-id=none -N -static-libgcc -lgcc
+ELFFLAGS	+=	-Wl,--build-id=none -N -static-libgcc
 ELFFLAGS	+=	$(platform-ldflags-y)
 ELFFLAGS	+=	$(firmware-ldflags-y)
 
-- 
2.25.1

